learning/tour-of-beam/frontend/lib/auth/notifier.dart

```diff
@@ -54,7 +56,8 @@ class AuthNotifier extends ChangeNotifier {
 
   Future<void> deleteAccount() async {
     try {
-      await client.postDeleteUserProgress();
+      // TODO(nausharipov) review: is there a better place for this?
```

Comment:
Add `AccountDeletionListener`, notify all listeners here.
Add `final _accountDeletionListeners = <AccountDeletionListener>[];`

With methods like `beforeAccountDeleted` or `afterAccountDeleted`. Likely we should do "before...". Those methods should be `async`, we should wait for them to complete.

Reply:
The implementation added extra code, which is not necessary for now. Added a comment.